plugins/resources/compound/libcompound.cpp

```diff
@@ -1853,7 +1853,7 @@ irods::error open_for_prefer_cache_policy(
 
     // =-=-=-=-=-=-=-
     // If the archive voted higher it will need to be staged to cache
-    if ( arch_check_vote > cache_check_vote ) {
+    if ( irods::WRITE_OPERATION != *_opr && arch_check_vote > cache_check_vote ) {
```

Comment:
I think we should add a test for the unnecessary data movement. Here's a scenario that clearly demonstrates it:
```bash
$ ils -l foo
  rods              3 comp0;archive0          240 2023-02-27.18:24 & foo
  rods              4 other          240 2023-02-27.18:22 & foo
$ echo "this other stuff" | istream -R other write foo
$ ils -l foo
  rods              3 comp0;archive0          240 2023-02-27.18:24 X foo
  rods              4 other           17 2023-02-27.18:24 & foo
  rods              5 comp0;cache0          240 2023-02-27.18:24 X foo
```
With your changes, we would expect nothing to appear in the cache, as I understand it.

It's a little harder to show that the unnecessary movement isn't happening when the compound resource is actually targeted for the write, though.

Thoughts?

Reply:
Test case added and pushed to 4-2-stable.  Please take a look and if it is good I will roll it into main.

Reply:
I agree. We should definitely capture that in a test.